pkg/storage/fs/checker.go

```diff
@@ -14,6 +14,14 @@ func (v *storageImpl) Exists(ctx context.Context, module, version string) (bool,
 	defer span.End()
 	versionedPath := v.versionLocation(module, version)
 
+	ok, err := afero.Exists(v.filesystem, versionedPath)
```

Comment:
is this needed? Can we not just return false if the `afero.ReadDir` below returned a "NotExists" error?